File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int left = 0, right = nums.length / 2;
        int[] auxiliary = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] >= 0) auxiliary[left++] = nums[i];
            else auxiliary[right++] = nums[i];
        }
        left = 0;
        right = nums.length / 2;
        for (int i = 0; i < auxiliary.length; i++) {
            if (i % 2 == 0) nums[i] = auxiliary[left++];
            else nums[i] = auxiliary[right++];
        }
        return nums;
    }
}

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int N = nums.size();
        
        vector<int> res(N);
        
        for (int i = 0, p = 0, n = 1; i < N; i++) {
            if (nums[i] >= 0) {
                res[p] = nums[i];
                p += 2;
            } else {
                res[n] = nums[i];
                n += 2;
            }
        }
        
        return res;
    }
};